internal/controllers/machine_controller.go

```diff
@@ -698,15 +724,19 @@ func (r *MachineReconciler) domainFor(
 	}
 
 	if machineImgRef := machine.Spec.Image; machineImgRef != nil && ptr.Deref(machineImgRef, "") != "" {
-		if err := r.setDomainImage(ctx, machine, domainDesc, ptr.Deref(machineImgRef, "")); err != nil {
+		if err := r.setDomainImage(ctx, log, machine, domainDesc, ptr.Deref(machineImgRef, "")); err != nil {
 			return nil, nil, nil, err
 		}
 	}
 
 	if ignitionSpec := machine.Spec.Ignition; ignitionSpec != nil {
-		if err := r.setDomainIgnition(machine, domainDesc); err != nil {
+		if err := r.setDomainIgnition(log, machine, domainDesc); err != nil {
 			return nil, nil, nil, err
 		}
+	} else {
+		if eventErr := r.eventStore.AddEvent(machine.Metadata, corev1.EventTypeWarning, "NoIgnitionData", "Machine does not have ignition data"); eventErr != nil {
```

Comment:
this event doesn't have dot on end of line. I personally prefer this format. How k8s doing that?

Reply:
Since Kubernetes follows the same convention, I adopted it as well.